flask_appbuilder/api/manager.py

```diff
@@ -58,22 +58,23 @@ def _create_api_spec(version):
             openapi_version="3.0.2",
             info=dict(description=current_app.appbuilder.app_name),
             plugins=[MarshmallowPlugin()],
-            servers=[{"url": "/api/{}".format(version)}],
+            servers=[{"url": url_for("OpenApi.get", version=version)[:-9]}],
```

Comment:
Sorry, what this for `[:-9]` ?

Reply:
url_for("OpenApi.get", version=version) returns url_root + /api/v1/_openapi but we just need url_root + /api/v1

Reply:
I’d use `.rpartition("/")[0]`, which splits the string at the last slash and returns everything before that point.

Isn’t there a better source for the API version root URL however?

Reply:
The class `OpenApi` itself holds the base and the version is passed to this method. The API version can vary on each `BaseApi` class so the source is at that level

Reply:
so what do we change?

Reply:
Make it

```python
servers=[{"url": url_for("OpenApi.get", version=version).rpartition("/")[0]}],
```

although I'd pull out the expression into a separate variable:

```python
    @staticmethod
    def _create_api_spec(version):
        # The spec produced here applies to the base of this view:
        api_url = url_for("OpenApi.get", version=version).rpartition("/")[0]
        return APISpec(
            title=current_app.appbuilder.app_name,
            version=version,
            openapi_version="3.0.2",
            info=dict(description=current_app.appbuilder.app_name),
            plugins=[MarshmallowPlugin()],
            servers=[{"url": api_url}],
        )
```

Reply:
i made the changes, how do i merge the change it into the pull request .. sometimes i dont understand github